core/Rename.cpp

```diff
@@ -142,10 +258,27 @@ namespace olympia
                     const auto rf  = determineRegisterFile(src);
                     const auto num = src.field_value;
                     auto & bitmask = renaming_inst->getSrcRegisterBitMask(rf);
-                    bitmask.set(num);
+                    uint32_t prf;
+
+                    bool no_prf = false;
+                    if(!map_table_[rf][num].second){
+                        prf = num;
+                        // we store the mapping, so when processing the retired instruction,
+                        // we know that there was no real mapping, so we mark it as no_prf
+                        no_prf = true;
+                    }
+                    else{
+                        prf = map_table_[rf][num].first;
+                        // only increase reference counter for real references
+                        reference_counter_[rf][prf]++;
+                    }
+                    olympia::Inst::RenameData::SourceReg rename_data_src(prf, no_prf);
+                    renaming_inst->getRenameData().setSource(rename_data_src);
+                    bitmask.set(prf);
+                    scoreboards_[rf]->set(bitmask);
```

Comment:
Wanna make sure if it's your desired behavior: this would make source PRFs ready in Rename. Which means, operand readiness are set by itself, not other dependent instructions.

Reply:
For this PR, I'm only implementing the renaming structures, but I'm not checking for operand dependencies yet, that'll be the next task. So I'm just marking all instructions as ready to allow them to execute.